engine/main.py

```diff
@@ -0,0 +1,47 @@
+import math
+import threading
+import json
+
+from constants.definitions import ENGINE_PATH
+from engine.robot import Robot
+from engine.robot_state import Robot_State
+from engine.mission import Mission
+from engine.mission_state import Mission_State
+from engine.control_mode import ControlMode
+from engine.database import DataBase
+from engine.transmission import send_packet_to_gui
+from engine.is_raspberrypi import is_raspberrypi
+from engine.plot_trajectory import plot_sim_traj
+from engine.parser import parse_main
+
+if __name__ == "__main__":
+    # Load config from JSON file
+    with open(ENGINE_PATH +"/config.json", "r") as fp:
+        config_args = json.load(fp)
+    
+    # Load user-defined arguments
+    user_args = parse_main()
+
+    # Development config flags
+    is_transmit = config_args.get("is_transmit") # Set to true when the rpi/robot is communicating w/ the GUI
+    is_sim = config_args.get("is_sim") if is_raspberrypi() else not is_raspberrypi() # Set to true when simulating the rpi, set to false when running on rpi
+    store_data = config_args.get("store_data") # Set to true when we want to track/store csv data
```

Comment:
if im not mistaken, this is only used to pass into Robot_State. I think it might be more concise to directly have this there?

Reply:
pushed a change for this. i was thinking about passing config arg into robot state but after testing i realized we would need to load the config args in every test which kinda sucks so i passed in the values of is_sim and store_data there not sure if thats the best approach tho?